NetCasbin/Abstractions/IEnforcer.cs

```diff
@@ -38,53 +38,53 @@ public interface IEnforcer : IManagementEnforcer
         /// <param name="name"></param>
         /// <param name="role"></param>
         /// <returns></returns>
-        bool HasRoleForUser(string name, string role);
+        bool HasRoleForUser(string name, string role, params string[] domain);
 
         /// <summary>
         /// Adds a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
         /// <returns>Returns false if the user already has the role (aka not affected).</returns>
-        bool AddRoleForUser(string user, string role);
+        bool AddRoleForUser(string user, string role, params string[] domain);
 
         /// <summary>
         /// Adds a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
         /// <returns>Returns false if the user already has the role (aka not affected).</returns>
-        Task<bool> AddRoleForUserAsync(string user, string role);
+        Task<bool> AddRoleForUserAsync(string user, string role, params string[] domain);
 
         /// <summary>
         /// Deletes a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
         /// <returns>Returns false if the user does not have the role (aka not affected).</returns>
-        bool DeleteRoleForUser(string user, string role);
+        bool DeleteRoleForUser(string user, string role, params string[] domain);
 
         /// <summary>
         /// Deletes a role for a user.
         /// </summary>
         /// <param name="user"></param>
         /// <param name="role"></param>
         /// <returns>Returns false if the user does not have the role (aka not affected).</returns>
-        Task<bool> DeleteRoleForUserAsync(string user, string role);
+        Task<bool> DeleteRoleForUserAsync(string user, string role, params string[] domain);
```

Comment:
@Sagilio Go Casbin uses []string domain because it's the only way to represent optional parameter in Golang. I think it's unnecessary in C#. Just use a single optional string arg.